shared/src/main/kotlin/org/javacs/kt/classpath/KScriptClassPathResolver.kt

```diff
@@ -0,0 +1,39 @@
+package org.javacs.kt.classpath
+
+import org.javacs.kt.LOG
+import java.nio.file.Path
+
+/** Resolver for reading maven dependencies */
+internal class KScriptClassPathResolver private constructor(private val script: Path) : ClassPathResolver {
+    override val resolverType: String = "KScript"
+    override val classpath: Set<Path> get() {
+        val artifacts = readKScriptDependencyList(script)
+
+        when {
+            artifacts.isEmpty() -> LOG.warn("No artifacts found in {}", script)
+            artifacts.size < 5 -> LOG.info("Found {} in {}", artifacts, script)
+            else -> LOG.info("Found {} artifacts in {}", artifacts.size, script)
+        }
+
+        return artifacts.mapNotNull { findMavenArtifact(it, false) }.toSet()
+    }
+
+    companion object {
+        /** Create a maven resolver if a file is a pom. */
+        fun maybeCreate(file: Path): KScriptClassPathResolver? =
+            file.takeIf {
+                it.toString().endsWith("kts") &&
+                    it.toFile().bufferedReader().use { b -> b.readLine().contains("kscript") }
+            }?.let { KScriptClassPathResolver(file) }
+    }
+}
+
+private val artifactPattern = "^//DEPS .*".toRegex()
+
+private fun readKScriptDependencyList(script: Path): Set<Artifact> =
+    script.toFile()
+        .readLines()
+        .filter { artifactPattern.matches(it) }
+        .flatMap { it.substring("//DEPS ".length).split(",") }
+        .map { parseMavenArtifact(it) }
```

Comment:
I'm a bit worried that this resolver is coupled too closely to Maven. What happens if e.g. a user uses Gradle instead? If we want to go this route, we might want to make it more generic, e.g. support different resolvers via `//DEPS maven:..., gradle:...` or even plain JARs.

Also I would probably go with a different prefix. `//DEPS` feels both a bit generic and magical at the same time. Perhaps we should mention the language server, i.e. something like

- `// kotlin-language-server: ...`
- `// kls-deps: ...`
- `// kls-classpath: ...`

or similar. The last option would have the advantage of being consistent with our [`ShellClassPathResolver`](https://github.com/fwcd/kotlin-language-server/blob/main/shared/src/main/kotlin/org/javacs/kt/classpath/ShellClassPathResolver.kt), which uses a file called [`kls-classpath`](https://github.com/fwcd/kotlin-language-server#figuring-out-the-dependencies) to resolve the dependencies.

Reply:
I think the same, consider that kscript already resolves the dependencies, so I think kscript should be used to obtain the dependencies, the problem would be that kscript does not yet have an option to return the path of these dependencies, this [PR](https://github.com/kscripting/kscript/pull/420) adds this function but is pending merging.

As a test, I created a [bash script](https://github.com/esturafd/ksdeps) to generate the kls-classpath file with this new feature of listing dependencies.

Another point would be the syntax of `//DEPS` is from kscript (or rather inherited from Jbang), but currently kscript has it deprecated, the currently recommended one would be [this](https://github.com/kscripting/kscript?tab=readme-ov-file#script-configuration).